src/test/fuzz/fuzz.cpp

```diff
@@ -25,6 +25,7 @@ static bool read_stdin(std::vector<uint8_t>& data)
 static void initialize()
 {
     const static auto verify_handle = MakeUnique<ECCVerifyHandle>();
+    SelectParams(CBaseChainParams::REGTEST);
```

Comment:
In commit 115bafe488ef0c53a88c2bc1e1b0508867c1daa8:

Not sure if we want to pick regtest for all tests. I'd argue to pick main.

Or maybe leave it to the individual tests to pick the params? This will result in a (slight) run time cost, but might make the code more clear?

Reply:
It not always entirely clear which functions that indirectly depend on params being selected and the requirement can be quite far down in the call graph.

An example is from when I fuzzed `EvalDescriptorStringOrObject`:

When using [this harness](https://github.com/bitcoin/bitcoin/blob/f27f6b0da9491e761274ba95fe15381524542bdf/src/test/fuzz/parse_univalue.cpp) the fuzzer managed to create the input string `"addr()"` after roughly three million executions. That input triggered the following assertion failure:

```
FUZZER: #3170783	REDUCE cov: 3338 ft: 7992 corp: 325/63Kb exec/s: 1258 rss: 272Mb L: 61/4090 MS: 2 ChangeASCIIInt-EraseBytes-
FUZZER: fuzzer: chainparams.cpp:385: const CChainParams &Params(): Assertion `globalChainParams' failed.
...
FUZZER: \"addr()\"
```

Turns out `EvalDescriptorStringOrObject` calls `Parse` which calls `DecodeDestination` which in turn depends on params being selected.

Since the requirement can run quite deep I think we should pick a network in the general `initialize()` so that harness writers don't have to worry about crossing any code paths with params requirements that will generate faux crashes.

OTOH, I don't feel strongly about the particular choice of network - I just choose `regtest` since that seemed like the least adventurous choice :)

What would be the pros/cons of going with `main` instead of going `regtest` (or `test`)?

Reply:
I'd rather want to test against a (clean-as-possible) mainnet, than a testnet. However, some tests might want to use segwit or lower difficulty blocks, which is enabled only on regtest from genesis.

I think the choice really depends on each individual test.

I see that you run into a particular issue with the chainparams, but similarly a fuzzer might find a code path that requires connman or a mempool, or anything else. In that case we'd have to initialize the full `(Basic)Testingsetup` here. Which in turn is not only expensive and wasteful to do, but also hard to reset for each input. So I'd reiterate my previous point that this should probably be picked for each test case either:
* Nothing at all
* A plain `SelectParams`
* An advanced `BasicTestingSetup`
* A full `TestingSetup`

Reply:
My fuzzing experience is that chainparams requirements are more scattered across the code base compared to connman and mempool requirements and thus more likely to trip on accidentally. Perhaps we can take the pick-level-of-initialization stuff in a follow-up PR?

Reply:
Let's wait to see other people's opinion on this

Reply:
Feedback addressed. Now allowing individual tests to pick the params. Picking `regtest` for this override since that is generally the least risky option in case of fuzzer gone haywire (think the scenario of a fuzzer writing to disk for some strange reason)  - also nice with low difficulty :)  Please re-review.